src/legacy/core_plugins/metrics/public/components/panel_config/timeseries.js

```diff
@@ -42,21 +42,28 @@ import {
   EuiHorizontalRule,
 } from '@elastic/eui';
 import { injectI18n, FormattedMessage } from '@kbn/i18n/react';
-
+import { Storage } from 'ui/storage';
+import { data } from 'plugins/data';
+import chrome from 'ui/chrome';
+const uiSettingsQueryLanguage = chrome.getUiSettingsClient().get('search:queryLanguage');
+const { QueryBarInput } = data.query.ui;
+const localStorage = new Storage(window.localStorage);
```

Comment:
@Bargs can we make '`store`' props optional for `QueryBarInput` and set `new Storage(window.localStorage)` as a default value.

My main concern in usage of `window.localStorage` Not sure that TSVB should know anything about types of storages in our app.

Or we can modify Storage constructor but it looks more risky.

@timroes @markov00 any ideas? 

_Optional for this PR._

Reply:
I wouldn't worry about it in this PR. In my filters agg PR I've already created [a service](https://github.com/elastic/kibana/pull/37287/files#diff-94b1fa464254dc42ebadd419f72fc85d) in plain JS that's similar to our angular service. Once that PR is merged we can update this code if we want to. Or if Tina would like she can cherry pick that change from my PR, but I'll leave that up to her.

Reply:
I'll update the code once the filters agg PR is merged.